### PR TITLE
Initial Model Training

### PR DESCRIPTION
**Pull request by mradovic38**
* Created a training notebook
* Implemented a flow model with U-Net architecture
* Split the dataset to train, val and test sets
* Created a training objective
* Upgrade trainer script
  * Implemented validation on validation dataset
  * Implemented model checkpointing (#2)
  * Introduced cosine learning rate scheduling (#3)
* Added gradient clipping to address exploding loss issue (#4)

### CODE REVIEW

**Comment by mradovic38** [comment]
* Added test loss calculation (#1)
* Updated val loss calculation (now goes over the entire val dataset in batches)

**Comment by mradovic38** [comment]
* Implement FID score using `torchmetrics`
* Use FID as validation metric
* Test-train the initial U-Net model
